### PR TITLE
fix: remove leading li when pasting inside a list

### PR DESCRIPTION
**Pull request by z0al**

### CODE REVIEW

File: packages/rich-text/src/plugins/List/insertListFragment.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+const unwrapTextContainerAtStart = <T extends TDescendant>(nodes: T[]): T[] => {
+  const node = nodes[0];
+
+  if (TEXT_CONTAINERS.includes(node.type)) {
+    return [...node.children, ...nodes.slice(1)];
+  }
+
+  return nodes;
+};
```
Comment (z0al) [inline packages/rich-text/src/plugins/List/insertListFragment.ts:72]: The actual fix. Everything else is mostly refactoring

File: packages/rich-text/src/plugins/List/__tests__/insertListFragment.test.tsx [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+    },
+    {
+      title: 'single li with only asset card',
+      input: (
+        <hul>
+          <hli>
+            <hp>
+              hello
+              <cursor />
+              world
```
Comment (DanweDE) [inline packages/rich-text/src/plugins/List/__tests__/insertListFragment.test.tsx:66]: I assume this is equal to `hello<cursor />world` without any soft line breaks, right?

File: packages/rich-text/src/plugins/List/__tests__/insertListFragment.test.tsx [hunk 3]
```diff
@@ -0,0 +1,200 @@
+/** @jsx jsx */
+import { jsx, assertOutput, createTestEditor } from '../../../test-utils';
+
+describe('insert fragment', () => {
```
Comment (DanweDE) [inline packages/rich-text/src/plugins/List/__tests__/insertListFragment.test.tsx:4]: Amazing set of new tests :+1:

**Review by DanweDE — APPROVED** [review]

File: packages/rich-text/src/plugins/List/__tests__/insertListFragment.test.tsx (hunk 2 above)
Comment (z0al) [reply packages/rich-text/src/plugins/List/__tests__/insertListFragment.test.tsx:66]: Yes. Prettier decided to format it this way

File: packages/rich-text/src/plugins/List/components/List.tsx [hunk 4]
```diff
@@ -8,10 +8,15 @@ import * as Slate from 'slate-react';
...
   padding: 0;
   margin: 0 0 1.25rem 1.25rem;
+
   div:first-child {
     margin: 0;
     line-height: ${tokens.lineHeightDefault};
   }
+
+  li + li {
+    margin-top: ${tokens.spacingXs};
```
Comment (DanweDE) [inline packages/rich-text/src/plugins/List/components/List.tsx:18]: What's the motivation for this one?

I actually noticed some odd spacing behavior playing around with this:
spacing-issues

**Review by DanweDE — CHANGES_REQUESTED** [review]
There seems to be a spacing issue actually (see comment above)

File: packages/rich-text/src/plugins/List/components/List.tsx (hunk 4 above)
Comment (z0al) [reply packages/rich-text/src/plugins/List/components/List.tsx:18]: Good catch. It was supposed to provide consistent spacing between li elements. I took it from the old editor code but looks like I broke something else. will check 👍🏻

Comment (z0al) [reply packages/rich-text/src/plugins/List/components/List.tsx]: pushed a fix

**Review by DanweDE — APPROVED** [review]

**Pull request merged**